src/lib/syncDocsPath.ts

```diff
@@ -127,6 +130,40 @@ async function pushDoc(
     });
 }
 
+function sortFiles(files: string[], { allowedFileExtensions }: { allowedFileExtensions: string[] }): string[] {
+  const extensionRegexp = new RegExp(`(${allowedFileExtensions.join('|')})$`);
+
+  const filesBySlug = files
+    .map(filePath => {
+      const doc = frontMatter(fsSync.readFileSync(filePath, 'utf8'));
+      const slug = path.basename(filePath).replace(extensionRegexp, '');
+      const parentDocSlug = doc.data.parentDocSlug;
+
+      return { filePath, slug, parentDocSlug };
+    })
+    .reduce(
+      (bySlug, obj) => {
+        // eslint-disable-next-line no-param-reassign
+        bySlug[obj.slug] = obj;
+        return bySlug;
+      },
+      {} as Record<string, { filePath: string; parentDocSlug: string; slug: string }>,
+    );
+
+  const dependencies = Object.values(filesBySlug).reduce(
+    (edges, obj) => {
+      if (obj.parentDocSlug) {
```

Comment:
i know we have users that sync only parts of their docs using `rdme`, i think we should probably also ensure that the parent file actually exists

(also let me know if i'm misunderstanding how this works!)

```suggestion
      if (obj.parentDocSlug && filesBySlug[obj.parentDocSlug]) {
```